crates/weaver_resolver/src/registry.rs

```diff
@@ -543,30 +583,77 @@ fn resolve_extends_references(ureg: &mut UnresolvedRegistry) -> Result<(), Error
         for unresolved_group in ureg.groups.iter_mut() {
             if let Some(extends) = unresolved_group.group.extends.as_ref() {
                 if let Some(attrs) = group_index.get(extends) {
-                    unresolved_group.attributes = resolve_inheritance_attrs(
+                    unresolved_group.attributes = resolve_inheritance_attrs_unified(
                         &unresolved_group.group.id,
                         &unresolved_group.attributes,
-                        extends,
-                        attrs,
+                        vec![(extends, attrs)],
                         unresolved_group.group.lineage.as_mut(),
                     );
-                    _ = unresolved_group.group.extends.take();
-                    _ = group_index.insert(
-                        unresolved_group.group.id.clone(),
-                        unresolved_group.attributes.clone(),
+                    add_resolved_group_to_index(
+                        &mut group_index,
+                        unresolved_group,
+                        &mut resolved_extends_count,
                     );
-                    resolved_extends_count += 1;
                 } else {
                     errors.push(Error::UnresolvedExtendsRef {
                         group_id: unresolved_group.group.id.clone(),
-                        extends_ref: unresolved_group
-                            .group
-                            .extends
-                            .clone()
-                            .unwrap_or("".to_owned()),
+                        extends_ref: extends.clone(),
                         provenance: unresolved_group.provenance.clone(),
                     });
                 }
+            } else if !unresolved_group.group.include_groups.is_empty() {
+                // Iterate over all groups and resolve the `include_groups` clauses.
+                let mut attr_ids = HashMap::new();
+                let mut attrs_by_group = HashMap::new();
+                let mut all_resolved = true;
+
+                for include_group in unresolved_group.group.include_groups.iter() {
+                    if let Some(attrs) = group_index.get(include_group) {
+                        // check if any of the attr in attrs is already in the all_attrs
+                        // and fail, otherwise add them to all_attrs
+                        for attr in attrs {
+                            if attr_ids.contains_key(&attr.spec.id()) {
+                                log::debug!(
+                                    "Found duplicate attribute id {} in include_groups for group {}",
+                                    attr.spec.id(),
+                                    unresolved_group.group.id
+                                );
+                                // TODO: accumulate real errors and return all of them
```

Comment:
I believe you can actually just:

```
errors.push(Error::DuplicateAttributeId { ... });
all_resolved = false;
```

And let this continue to the error reporting below.